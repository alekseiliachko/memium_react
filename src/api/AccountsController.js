import APIController from "./ApiController";

class АccountController extends APIController {
  constructor() {
    super();
  }

  async getAvatarImage() {
    const avatar = await this.request("get", "/account/avatar", {
      responseType: "blob",
    });
    return URL.createObjectURL(avatar.data);
  }

  async updateAvatar(avatar) {
    const updatedAvatar = await this.request(
      "post",
      `/account/avatar`,
      avatar,
      {
        responseType: "blob",
      }
    );
    return URL.createObjectURL(updatedAvatar.data);
  }

  async getUserDetails() {
    const details = await this.request("get", "/account/details");
    const fetchedDetails = Object.entries(details.data).map(([key, value]) => {
      if (value == null) {
        return [key, ""];
      }
      return [key, value];
    });
    return Object.fromEntries(fetchedDetails);
  }

  async getBlUsers() {
    const users = await this.request("get", "/account/bl");
    return users;
  }

  async deleteUserFromBl(userID) {
    const deleteStatus = await this.request("delete", `/account/bl/${userID}`);
    return deleteStatus;
  }

  async getLikedPosts() {
    const posts = await this.request("get", "/account/like");
    return posts;
  }

  async deleteLikeFromPost(articleId) {
    const deleteStatus = await this.request(
      "delete",
      `/account/like/${articleId}`
    );
    return deleteStatus;
  }

  async getRecommendedUsers() {
    const users = await this.request("get", "/account/home/");
    return users;
  }

  async getSubscribersList() {
    const users = await this.request("get", "/account/sub");
    return users;
  }

  async updateSubscribersList(userID) {
    const updateStatus = await this.request("post", `/account/${userID}/like/`);
    return updateStatus;
  }

  async updateDetails(details) {
    const updatedDetails = await this.request(
      "put",
      `/account/details`,
      details
    );
    return updatedDetails;
  }
}

export default new АccountController();
