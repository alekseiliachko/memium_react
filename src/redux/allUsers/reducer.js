import { SET_USER_AVATAR, SET_USER_DATA } from "./actions";

export const initialState = {
  userDataById: {},
};

export const allUsersReducer = (store = initialState, action) => {
  switch (action.type) {
    case SET_USER_DATA:
      return {
        ...store,
        userDataById: {
          ...store.userDataById,
          [action.userId]: {
            ...store.userDataById[action.userId],
            ...action.userData,
          },
        },
      };
    case SET_USER_AVATAR:
      return {
        ...store,
        userDataById: {
          ...store.userDataById,
          [action.userId]: {
            ...store.userDataById[action.userId],
            avatar: action.avatar,
          },
        },
      };
    default:
      return store;
  }
};
