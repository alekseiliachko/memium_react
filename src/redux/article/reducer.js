import { RESET_ARTICLE_DATA, SET_ARTICLE_DATA } from "./actions";

const initialState = {
  articleDataById: {},
};

export const articleReducer = (store = initialState, action) => {
  switch (action.type) {
    case SET_ARTICLE_DATA:
      return {
        ...store,
        articleDataById: {
          ...store.articleDataById,
          [action.articleId]: action.articleData,
        },
      };
    case RESET_ARTICLE_DATA:
      return {
        ...store,
        articleDataById: {
          ...store.articleDataById,
          [action.articleId]: undefined,
        },
      };
    default:
      return store;
  }
};
