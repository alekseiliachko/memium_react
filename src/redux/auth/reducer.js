import { LOGIN_ATTEMPT, LOGIN_SUCCESS, LOG_OUT } from "./actions";

export const AUTH_STATE = {
  LOGGED_OUT: "LOGGED_OUT",
  SUCCESS: "SUCCESS",
  ATTEMPT: "ATTEMPT",
};

const initialState = {
  username: "",
  authState: AUTH_STATE.ATTEMPT,
};

export const authReducer = (store = initialState, action) => {
  switch (action.type) {
    case LOGIN_ATTEMPT:
      return {
        ...store,
        authState: AUTH_STATE.ATTEMPT,
      };
    case LOGIN_SUCCESS:
      return {
        ...store,
        username: action.payload,
        authState: AUTH_STATE.SUCCESS,
      };
    case LOG_OUT:
      return {
        ...store,
        username: "",
        authState: AUTH_STATE.LOGGED_OUT,
      };
    default:
      return store;
  }
};
