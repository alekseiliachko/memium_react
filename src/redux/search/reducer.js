import { UPDATE_SEARCH_RESULTS } from "./actions";

const initialState = {
  articles: [],
  authors: [],
};

export const searchReducer = (store = initialState, action) => {
  switch (action.type) {
    case UPDATE_SEARCH_RESULTS:
      return {
        ...store,
        articles: action.articles,
        authors: action.authors,
      };
    default:
      return store;
  }
};
