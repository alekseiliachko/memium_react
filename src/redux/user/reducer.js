import {
  AVATAR_LOADED,
  ATTEMPT_LOAD_AVATAR,
  ATTEMPT_LOAD_DETAILS,
  DETAILS_LOADED,
  SUBS_LOADED,
  ATTEMPT_LOAD_SUBS,
  ATTEMPT_UPDATE_DETAILS,
  ATTEMPT_LOAD_BL,
  BL_LOADED,
  LIKED_LOADED,
  ATTEMPT_LOAD_LIKED,
} from "./actions";

export const LOADING_STATUS = {
  NOT_LOADED: "NOT_LOADED",
  LOADING: "LOADING",
  LOADED: "LOADED",
};

const initialState = {
  avatar: "",
  blackList: [],
  details: {
    bio: "",
    gender: "Another",
    name: "",
  },
  subs: [],
  likedList: [],
  avatarLoadingStatus: LOADING_STATUS.NOT_LOADED,
  detailsLoadingStatus: LOADING_STATUS.NOT_LOADED,
  subsLoadingStatus: LOADING_STATUS.NOT_LOADED,
  blackListLoadingStatus: LOADING_STATUS.NOT_LOADED,
  likedLoadingStatus: LOADING_STATUS.NOT_LOADED,
};

export const userReducer = (store = initialState, action) => {
  switch (action.type) {
    case AVATAR_LOADED:
      return {
        ...store,
        avatar: action.payload,
        avatarLoadingStatus: LOADING_STATUS.LOADED,
      };
    case ATTEMPT_LOAD_AVATAR:
      return {
        ...store,
        avatarLoadingStatus: LOADING_STATUS.LOADING,
      };
    case DETAILS_LOADED:
      return {
        ...store,
        details: action.payload,
        detailsLoadingStatus: LOADING_STATUS.LOADED,
      };
    case ATTEMPT_LOAD_DETAILS:
      return {
        ...store,
        detailsLoadingStatus: LOADING_STATUS.LOADING,
      };
    case SUBS_LOADED:
      return {
        ...store,
        subs: action.payload,
        subsLoadingStatus: LOADING_STATUS.LOADED,
      };
    case ATTEMPT_LOAD_SUBS:
      return {
        ...store,
        subsLoadingStatus: LOADING_STATUS.LOADING,
      };
    case ATTEMPT_UPDATE_DETAILS:
      return {
        ...store,
        detailsLoadingStatus: LOADING_STATUS.LOADING,
      };
    case BL_LOADED:
      return {
        ...store,
        blackList: action.payload,
        blackListLoadingStatus: LOADING_STATUS.LOADED,
      };
    case ATTEMPT_LOAD_BL:
      return {
        ...store,
        blackListLoadingStatus: LOADING_STATUS.LOADING,
      };
    case LIKED_LOADED:
      return {
        ...store,
        likedList: action.payload,
        likedListLoadingStatus: LOADING_STATUS.LOADED,
      };
    case ATTEMPT_LOAD_LIKED:
      return {
        ...store,
        likedListLoadingStatus: LOADING_STATUS.LOADING,
      };
    default:
      return store;
  }
};
