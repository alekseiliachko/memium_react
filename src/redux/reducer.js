import { combineReducers, createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { authReducer } from "./auth/reducer";
import { userReducer } from "./user/reducer";
import { articleReducer } from "./article/reducer";
import { allUsersReducer } from "./allUsers/reducer";
import { searchReducer } from "./search/reducer";

const rootState = combineReducers({
  authReducer,
  userReducer,
  articleReducer,
  allUsersReducer,
  searchReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware = [thunk];
export function configureStore() {
  const store = createStore(
    rootState,
    composeEnhancers(applyMiddleware(...middleware))
  );
  return store;
}
