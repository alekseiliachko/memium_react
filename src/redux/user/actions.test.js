import configureMockStore from "redux-mock-store";
import { initialState } from "./reducer";
import {
  avatarLoadingAttempt,
  blLoadingAttempt,
  deleteFromBl,
  detailsLoadingAttempt,
  detailsUpdateAttempt,
  feedLoadingAttempt,
  likedLoadingAttempt,
  loadAvatar,
  loadBl,
  loadDetails,
  loadLikedList,
  loadSubs,
  myArticlesLoadingAttempt,
  pushToLiked,
  setLoadedArticles,
  setLoadedAvatar,
  setLoadedBl,
  setLoadedDetails,
  setLoadedFeed,
  setLoadedLikedList,
  setLoadedSubs,
  subsLoadingAttempt,
  updateAvatar,
  updateDetails,
} from "./actions";
import thunk from "redux-thunk";
import AccountsController from "../../api/AccountsController";
jest.mock("../../api/AccountsController");

const mockStore = configureMockStore([thunk]);

const article = {
  id: "9064e587-3c29-4547-a5a6-51a8149456f8",
  imageUrl: null,
  authorId: "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
  title: "Русские мемчанские",
  description: null,
  date: 1615046988090,
  category: "Anime",
};

const account = {
  accountId: "22bb3fe5-185c-4c87-8633-1586586506b7",
  username: "zilber",
  name: null,
  bio: "populated",
  imageData: null,
};
describe("actions", () => {
  let store;
  beforeEach(() => {
    store = mockStore(initialState);
  });
  it("should create AVATAR_LOADED action", () => {
    store.dispatch(
      setLoadedAvatar(
        "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da"
      )
    );
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create DETAILS_LOADED action", () => {
    store.dispatch(setLoadedDetails(account));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create SUBS_LOADED action", () => {
    store.dispatch(setLoadedSubs([account]));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create BL_LOADED action", () => {
    store.dispatch(setLoadedBl([account]));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create FEED_LOADED action", () => {
    store.dispatch(
      setLoadedFeed({
        articles: [],
        accounts: null,
      })
    );
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create LIKED_LOADED action", () => {
    store.dispatch(setLoadedLikedList([article]));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create PUSH_TO_LIKED action", () => {
    store.dispatch(pushToLiked(article));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create MY_ARTICLES_LOADED action", () => {
    store.dispatch(setLoadedArticles([article]));
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create ATTEMPT_LOAD_AVATAR action", () => {
    store.dispatch(avatarLoadingAttempt());
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create ATTEMPT_LOAD_DETAILS action", () => {
    store.dispatch(detailsLoadingAttempt());
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create ATTEMPT_LOAD_SUBS action", () => {
    store.dispatch(subsLoadingAttempt());
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create ATTEMPT_UPDATE_DETAILS action", () => {
    store.dispatch(detailsUpdateAttempt());
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create ATTEMPT_LOAD_BL action", () => {
    store.dispatch(blLoadingAttempt());
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create ATTEMPT_LOAD_LIKED action", () => {
    store.dispatch(likedLoadingAttempt());
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create ATTEMPT_LOAD_FEED action", () => {
    store.dispatch(feedLoadingAttempt());
    expect(store.getActions()).toMatchSnapshot();
  });
  it("should create ATTEMPT_LOAD_MY_ARTICLES action", () => {
    store.dispatch(myArticlesLoadingAttempt());
    expect(store.getActions()).toMatchSnapshot();
  });
});

describe("async tests", () => {
  let store;
  beforeEach(() => {
    store = mockStore({ initialState });
  });

  it("create ATTEMPT_LOAD_AVATAR, AVATAR_LOADED when avatar loaded", () => {
    AccountsController.getAvatarImage.mockResolvedValue(
      "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da"
    );
    return store.dispatch(loadAvatar()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it("create ATTEMPT_LOAD_AVATAR, AVATAR_LOADED when avatar updated", () => {
    AccountsController.updateAvatar.mockResolvedValue(
      "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da"
    );
    return store
      .dispatch(
        updateAvatar(
          "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da"
        )
      )
      .then(() => {
        expect(store.getActions()).toMatchSnapshot();
      });
  });
  it("create ATTEMPT_LOAD_DETAILS, DETAILS_LOADED when loading details", () => {
    AccountsController.getUserDetails.mockResolvedValue(account);
    return store.dispatch(loadDetails()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it("create ATTEMPT_LOAD_SUBS, SUBS_LOADED when loading subs", () => {
    AccountsController.getSubscribersList.mockResolvedValue({
      data: [account],
    });
    return store.dispatch(loadSubs()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it("create ATTEMPT_UPDATE_DETAILS, DETAILS_LOADED when updating details", () => {
    AccountsController.updateDetails.mockResolvedValue({ data: account });
    return store.dispatch(updateDetails(account)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it("create ATTEMPT_LOAD_BL, BL_LOADED when loading blacklist", () => {
    AccountsController.getBlUsers.mockResolvedValue({ data: [account] });
    return store.dispatch(loadBl()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it("create BL_LOADED when delete from blacklist", () => {
    AccountsController.deleteUserFromBl.mockResolvedValue();
    AccountsController.getBlUsers.mockResolvedValue([]);
    return store.dispatch(deleteFromBl(account.accountId)).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
  it("create ATTEMPT_LOAD_LIKED, LIKED_LOADED when loading liked", () => {
    AccountsController.getLikedPosts.mockResolvedValue({ data: [article] });
    return store.dispatch(loadLikedList()).then(() => {
      expect(store.getActions()).toMatchSnapshot();
    });
  });
});
