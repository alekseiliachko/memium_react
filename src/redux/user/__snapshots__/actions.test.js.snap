// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions should create ATTEMPT_LOAD_AVATAR action 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_AVATAR",
  },
]
`;

exports[`actions should create ATTEMPT_LOAD_BL action 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_BL",
  },
]
`;

exports[`actions should create ATTEMPT_LOAD_DETAILS action 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_DETAILS",
  },
]
`;

exports[`actions should create ATTEMPT_LOAD_FEED action 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_FEED",
  },
]
`;

exports[`actions should create ATTEMPT_LOAD_LIKED action 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_LIKED",
  },
]
`;

exports[`actions should create ATTEMPT_LOAD_MY_ARTICLES action 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_MY_ARTICLES",
  },
]
`;

exports[`actions should create ATTEMPT_LOAD_SUBS action 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_SUBS",
  },
]
`;

exports[`actions should create ATTEMPT_UPDATE_DETAILS action 1`] = `
Array [
  Object {
    "type": "ATTEMPT_UPDATE_DETAILS",
  },
]
`;

exports[`actions should create AVATAR_LOADED action 1`] = `
Array [
  Object {
    "payload": "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da",
    "type": "AVATAR_LOADED",
  },
]
`;

exports[`actions should create BL_LOADED action 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "bio": "populated",
        "imageData": null,
        "name": null,
        "username": "zilber",
      },
    ],
    "type": "BL_LOADED",
  },
]
`;

exports[`actions should create DETAILS_LOADED action 1`] = `
Array [
  Object {
    "payload": Object {
      "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
      "bio": "populated",
      "imageData": null,
      "name": null,
      "username": "zilber",
    },
    "type": "DETAILS_LOADED",
  },
]
`;

exports[`actions should create FEED_LOADED action 1`] = `
Array [
  Object {
    "payload": Object {
      "accounts": null,
      "articles": Array [],
    },
    "type": "FEED_LOADED",
  },
]
`;

exports[`actions should create LIKED_LOADED action 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "category": "Anime",
        "date": 1615046988090,
        "description": null,
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "title": "Русские мемчанские",
      },
    ],
    "type": "LIKED_LOADED",
  },
]
`;

exports[`actions should create MY_ARTICLES_LOADED action 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "category": "Anime",
        "date": 1615046988090,
        "description": null,
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "title": "Русские мемчанские",
      },
    ],
    "type": "MY_ARTICLES_LOADED",
  },
]
`;

exports[`actions should create PUSH_TO_LIKED action 1`] = `
Array [
  Object {
    "payload": Object {
      "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
      "category": "Anime",
      "date": 1615046988090,
      "description": null,
      "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
      "imageUrl": null,
      "title": "Русские мемчанские",
    },
    "type": "PUSH_TO_LIKED",
  },
]
`;

exports[`actions should create SUBS_LOADED action 1`] = `
Array [
  Object {
    "payload": Array [
      Object {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "bio": "populated",
        "imageData": null,
        "name": null,
        "username": "zilber",
      },
    ],
    "type": "SUBS_LOADED",
  },
]
`;

exports[`async tests create ATTEMPT_LOAD_AVATAR, AVATAR_LOADED when avatar loaded 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_AVATAR",
  },
  Object {
    "payload": "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da",
    "type": "AVATAR_LOADED",
  },
]
`;

exports[`async tests create ATTEMPT_LOAD_AVATAR, AVATAR_LOADED when avatar updated 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_AVATAR",
  },
  Object {
    "payload": "blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da",
    "type": "AVATAR_LOADED",
  },
]
`;

exports[`async tests create ATTEMPT_LOAD_BL, BL_LOADED when loading blacklist 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_BL",
  },
  Object {
    "payload": Array [
      Object {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "bio": "populated",
        "imageData": null,
        "name": null,
        "username": "zilber",
      },
    ],
    "type": "BL_LOADED",
  },
]
`;

exports[`async tests create ATTEMPT_LOAD_DETAILS, DETAILS_LOADED when loading details 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_DETAILS",
  },
  Object {
    "payload": Object {
      "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
      "bio": "populated",
      "imageData": null,
      "name": null,
      "username": "zilber",
    },
    "type": "DETAILS_LOADED",
  },
]
`;

exports[`async tests create ATTEMPT_LOAD_LIKED, LIKED_LOADED when loading liked 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_LIKED",
  },
  Object {
    "payload": Array [
      Object {
        "authorId": "5d48bb9e-f1d9-41d2-bb02-9c4f578bd282",
        "category": "Anime",
        "date": 1615046988090,
        "description": null,
        "id": "9064e587-3c29-4547-a5a6-51a8149456f8",
        "imageUrl": null,
        "title": "Русские мемчанские",
      },
    ],
    "type": "LIKED_LOADED",
  },
]
`;

exports[`async tests create ATTEMPT_LOAD_SUBS, SUBS_LOADED when loading subs 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_SUBS",
  },
  Object {
    "payload": Array [
      Object {
        "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
        "bio": "populated",
        "imageData": null,
        "name": null,
        "username": "zilber",
      },
    ],
    "type": "SUBS_LOADED",
  },
]
`;

exports[`async tests create ATTEMPT_UPDATE_DETAILS, DETAILS_LOADED when updating details 1`] = `
Array [
  Object {
    "type": "ATTEMPT_UPDATE_DETAILS",
  },
  Object {
    "payload": Object {
      "accountId": "22bb3fe5-185c-4c87-8633-1586586506b7",
      "bio": "populated",
      "imageData": null,
      "name": null,
      "username": "zilber",
    },
    "type": "DETAILS_LOADED",
  },
]
`;

exports[`async tests create BL_LOADED when delete from blacklist 1`] = `
Array [
  Object {
    "type": "ATTEMPT_LOAD_BL",
  },
  Object {
    "payload": Array [],
    "type": "BL_LOADED",
  },
]
`;
